tendermint/src/lite_impl/signed_header.rs

```diff
@@ -52,68 +48,89 @@ impl lite::Commit for block::signed_header::SignedHeader {
-                    if vals.validator(precommit.validator_address) == None {
-                        let reason = format!(
-                            "Found a faulty signer ({}) not present in the validator set ({})",
-                            precommit.validator_address,
-                            vals.hash()
-                        );
-                        fail!(Kind::FaultyFullNode, reason);
-                    }
+        for commit_sig in self.commit.signatures.iter() {
+            // returns FaultyFullNode error if it detects a signer that is not present in the validator set
+            if let Some(val_addr) = commit_sig.validator_address {
+                if vals.validator(val_addr) == None {
+                    let reason = format!(
+                        "Found a faulty signer ({}) not present in the validator set ({})",
+                        val_addr,
+                        vals.hash()
+                    );
+                    fail!(Kind::FaultyFullNode, reason);
                 }
-                None => (),
             }
         }
 
         Ok(())
     }
 }
 
+fn commit_to_votes(commit: block::Commit) -> Result<Vec<vote::Vote>, Error> {
+    let mut votes: Vec<vote::Vote> = Default::default();
+    for (i, commit_sig) in commit.signatures.iter().enumerate() {
+        if commit_sig.is_absent() {
+            continue;
+        }
+
+        match commit_sig.validator_address {
+            Some(val_addr) => {
+                if let Some(sig) = commit_sig.signature.clone() {
+                    let vote = vote::Vote {
+                        vote_type: vote::Type::Precommit,
+                        height: commit.height,
+                        round: commit.round,
+                        block_id: Option::from(commit.block_id.clone()),
+                        timestamp: commit_sig.timestamp,
+                        validator_address: val_addr,
+                        validator_index: i as u64,
+                        signature: sig,
+                    };
+                    votes.push(vote);
+                }
+            }
+            None => {
+                fail!(
+                    Kind::ImplementationSpecific,
+                    "validator address missing in commit_sig {:#?}",
+                    commit_sig
+                );
+            }
+        }
+    }
+    Ok(votes)
+}
+
 impl block::signed_header::SignedHeader {
     /// This is a private helper method to iterate over the underlying
     /// votes to compute the voting power (see `voting_power_in` below).
-    fn iter(&self) -> Vec<Option<vote::SignedVote>> {
+    fn iter(&self) -> Result<Vec<vote::SignedVote>, Error> {
         let chain_id = self.header.chain_id.to_string();
-        let mut votes = self.commit.precommits.clone().into_vec();
-        votes
+        // if let Ok(mut votes) = commit_to_votes(self.commit.clone())
```

Comment:
?

Reply:
Seems to me that it was just refactored and maybe that line can go. Shivani?

Reply:
Oh yeah, I had kept that for a reference and then forgot to delete it. Thanks!